knowledge-base/persisting-the-root-certificate-in-fiddler-core.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Persisting the Root Certificate in Fiddler Core
+description: Article showing different methods for persisting the Certificate in Fiddler Core
+type: how-to
+page_title: Reusing Certificates in Fiddler Core
+slug: persisting-the-root-certificate-in-fiddler-core
+position: 
+tags: Certificates, Bouncy Castle, MakeCert
+ticketid: 1429862
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product Version</td>
+			<td>4.6.20191.7809</td>
+		</tr>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® FiddlerCore Embedded Engine</td>
+		</tr>
+	</tbody>
+</table>
+
+## Description
+FiddlerCore uses different APIs for certificate creation than the Desktop Version. By default, FiddlerCore includes the CertMaker and BCMakeCert assemblies for use with the Bouncy Castle API but doesn't persist the certificate by default. The Bouncy Castle API is recommended because it can be used across multiple platforms.
+
+## Suggested Workarounds
+Since FiddlerCore is a library that can be used in other applications, the API won't force persisting the Certificate and leaves the implementation up to the developer. The following workarounds are examples that show different ways this can be achieved.
+
+### Using MakeCert.dll with Application (Recommended)
+In this example, the MakeCert.dll is used with the ICertificateProvider5 Interface to store the Certificate Information. For example, call the `EnsureRootCertificate` method like below.
+
+``` csharp
+	private static void EnsureRootCertificate()
+	{
+		string certMakerPath = Path.Combine(assemblyDirectory, "lib", "CertMaker.dll");
```

Comment:
Let's use BCCertMaker here, as in the Capture Traffic demo: https://github.com/telerik/fiddler-core-demos/blob/master/CaptureTraffic/NetStandard/CaptureTraffic/Program.cs#L144

Reply:
Updated to use BCCertMaker implementation.